exchanges/orderbook/tranches.go

```diff
@@ -78,23 +78,25 @@ updates:
 
 // deleteByID deletes reference by ID
 func (ts *Tranches) deleteByID(updts Tranches, bypassErr bool) error {
+updates:
 	for x := range updts {
-		idx := slices.IndexFunc(*ts, func(t Tranche) bool {
-			return t.ID == updts[x].ID
-		})
+		for y := range *ts {
+			if updts[x].ID != (*ts)[y].ID {
+				continue
+			}
 
-		if idx == -1 {
-			if !bypassErr {
-				return fmt.Errorf("delete error: %w %d not found", errIDCannotBeMatched, updts[x].ID)
+			if y < len(*ts) {
+				copy((*ts)[y:], (*ts)[y+1:])
+				*ts = (*ts)[:len(*ts)-1]
+			} else {
+				*ts = slices.Delete(*ts, y, y+1)
 			}
```

Comment:
🚧 I do not understand this code. I feel that I must be being really stupid, except ... I don't think I am.

* y is an index range of *ts
* So `y < len(*ts)` is always true? The last value for y will be `len(*ts) - 1`

```suggestion
			copy((*ts)[y:], (*ts)[y+1:])
			*ts = (*ts)[:len(*ts)-1]
```